extensions/elation/actions/createCareGap/createCareGap.ts

```diff
@@ -0,0 +1,47 @@
+import { Category, type Action } from '@awell-health/extensions-core'
+import { makeAPIClient } from '../../client'
+import { type settings } from '../../settings'
+import { dataPoints, fields, FieldsValidationSchema } from './config'
+
+export const createCareGap: Action<
+  typeof fields,
+  typeof settings,
+  keyof typeof dataPoints
+> = {
+  key: 'createCareGap',
+  category: Category.EHR_INTEGRATIONS,
+  title: 'Create care gap',
+  description: 'Create a care gap in Elation.',
+  fields,
+  previewable: true,
+  dataPoints,
+  onActivityCreated: async (payload, onComplete, onError): Promise<void> => {    
+    const { patient_id, practice_id, created_date, status, detail, quality_program, definition_id } = FieldsValidationSchema.parse(payload.fields)
+
+    // The care gap API uses a different base URL than the rest of the Elation API, see https://docs.elationhealth.com/reference/caregaps_post_caregaps_api__quality_program__caregap__post-1
+    const baseUrl = payload.settings.base_url ?? ''
```

Comment:
What about using [this pattern](https://github.com/awell-health/awell-extensions/blob/main/extensions/elation/actions/getPatient/getPatient.ts#L22)? Anyway, I'd make sure to always validate the settings with the corresponding schema over accessing it directly via the `payload`.

```
    const {
      fields: {
        patient_id,
        practice_id,
        created_date,
        status,
        detail,
        quality_program,
        definition_id,
      },
      settings,
    } = validate({
      schema: z.object({
        fields: FieldsValidationSchema,
        settings: SettingsValidationSchema,
      }),
      payload,
    })
```

Reply:
Good suggestion. Updated.